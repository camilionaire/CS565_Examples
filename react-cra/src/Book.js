import React from 'react';
import './index.css';

export default function Book(props) {
  let list = props.data.map((item, index) => {
    return (
      <li key={index} className='book'>
        {item.name}
      </li>
    );
  });

  return <ol className='booklist'>{list}</ol>;
}
