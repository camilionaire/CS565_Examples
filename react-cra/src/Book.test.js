import React from 'react';
import { shallow, mount, render } from './enzyme';
import renderer from 'react-test-renderer';

import Book from './Book';

describe('<Book />', () => {
  it('debug', () => {
    const books = [];
    const wrapper = shallow(<Book data={books} />);
    console.log(wrapper.debug());
  });

  it('should check whether the className book exists', () => {
    const books = [];
    const wrapper = shallow(<Book data={books} />);
    expect(wrapper.find('.booklist').exists()).toBe(true);
  });

  it('renders correct text in books prop', () => {
    const books = ['book1', 'book2', 'book3'];
    const wrapper = mount(<Book data={books} />);
    console.log(wrapper.debug());
  });

  it('renders correctly', () => {
    const books = ['book1', 'book2', 'book3'];
    const tree = renderer.create(<Book data={books} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
