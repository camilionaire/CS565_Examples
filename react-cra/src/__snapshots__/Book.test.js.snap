// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> renders correctly 1`] = `
<ol
  className="booklist"
>
  <li
    className="book"
  />
  <li
    className="book"
  />
  <li
    className="book"
  />
</ol>
`;
